server/common/server-shared/src/main/java/org/kaaproject/kaa/server/sync/ClientSync.java

```diff
@@ -246,6 +244,22 @@ public BootstrapClientSync getBootstrapSync() {
     public void setBootstrapSync(BootstrapClientSync bootstrapSync) {
         this.bootstrapSync = bootstrapSync;
     }
+    
+    public boolean isValid() {
+        ClientSyncMetaData md = this.getClientSyncMetaData();
+        // TODO: validate if public key hash matches hash of public key during
+        // profile registration command.
+        if (md.getProfileHash() == null) {
```

Comment:
Slightly less code:
if (md.getProfileHash() == null) {
    ProfileClientSync profileRequest = this.getProfileSync();
    if (profileRequest == null || profileRequest.getEndpointPublicKey() == null) {
        return false;
    }
}
return true;